pkg/prometheus/operator.go

```diff
@@ -1403,7 +1403,33 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 		return errors.Wrap(err, "creating tls asset secret failed")
 	}
 
-	if err := c.createOrUpdateWebConfigSecret(ctx, p); err != nil {
```

Comment:
Any reason why we should drop the `createOrUpdateWebConfigSecret()` method? It makes `sync()` longer and harder to read.

Reply:
@simonpasquier I was trying to extract `createOrUpdateWebConfigSecret` function to share it between Prometheus and Alertmanager as what you suggested earlier. But IMHO, since we need quite a few information from either prometheus or alertmanager package, currently I don't have a good way to extract this function. Would you give me more hint on how to do this? You may check my earlier [commit](https://github.com/prometheus-operator/prometheus-operator/pull/4868/commits/2b4cb2fa97a2c8089194e99420402f960d834b84#diff-74002036bea7ef8f78c7623abba8d63a3e2578be73fb2dc8bcca154df57ce362R1655) for this part. Thanks :).

Reply:
My comment was more about keeping the `*Operator.createOrUpdateWebConfigSecret()` method in this file instead of inlining it in the `sync()` method as this current version of the change does.